cmd/allocator/main.go

```diff
@@ -214,12 +222,29 @@ func main() {
 	// This will test the connection to agones on each readiness probe
 	// so if one of the allocator pod can't reach Kubernetes it will be removed
 	// from the Kubernetes service.
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	podReady = true
 	health.AddReadinessCheck("allocator-agones-client", func() error {
+		if !podReady {
+			return errors.New("asked to shut down, failed readiness check")
+		}
 		_, err := agonesClient.ServerVersion()
-		return err
+		if err != nil {
+			return errors.New("failed to reach Kubernetes")
```

Comment:
you squashed the error from the client here - what was wrong with line 219? If you want to differentiate between these error cases you could do something like `fmt.Errorf("failed to reach Kubernetes: %w")`